Tests/mocks/Dobby.h

```diff
@@ -0,0 +1,294 @@
+#pragma once
+
+#include <list>
+#include <memory>
+#include <thread>
+#include <mutex>
+#include <set>
+#include <map>
+#include <functional>
+#include <condition_variable>
+#include <deque>
+
+
+#if defined(DOBBY_SERVICE_OVERRIDE)
+    #define DOBBY_SERVICE   DOBBY_SERVICE_OVERRIDE
+#else
+    #define DOBBY_SERVICE   "org.rdk.dobby"
+#endif
+#if defined(DOBBY_OBJECT_OVERRIDE)
+    #define DOBBY_OBJECT   DOBBY_OBJECT_OVERRIDE
+#else
+    #define DOBBY_OBJECT   "/org/rdk/dobby"
+#endif
+
+namespace AI_IPC
+{
+
+class IIpcService
+{
+public:
+    virtual ~IIpcService() = default;
+    virtual bool isValid() const = 0;
+    virtual bool isServiceAvailable(const std::string& serviceName) const = 0;
+    virtual void flush() = 0;
+    virtual bool start() = 0;
+    virtual bool stop() = 0;
+    virtual std::string getBusAddress() const = 0;
+};
+
+std::shared_ptr<IIpcService> createIpcService(const std::string& address, const std::string& serviceName, int defaultTimeoutMs = -1);
+
+} // AI_IPC
+
+
+class IpcService : public AI_IPC::IIpcService
+                 , public std::enable_shared_from_this<IpcService>
+{
+
+public:
+    static IpcService& getInstance()
+    {
+        static IpcService instance;
+        return instance;
+    }
+
+    IIpcService* impl;
+
+    IpcService(const std::string& dbusAddress, const std::string& serviceName, int defaultTimeoutMs = -1){
+
+    }
+
+    IpcService(){
+        IpcService("test", "test");
+    }
```

Comment:
Why are you creating an instance of class inside class constructor? Same in DobbyProxy().